### DISCUSSION TITLE
How to create a PWA Next.js application

### DISCUSSION THREAD

**Discussion by inezabonte** [vercel/next.js#24112] [Help] [answered]
I'm having a hard time finding sources that can help explain how PWAs are created with Next.js, I would really appreciate if anyone has sources that can help.

**Comment by leerob** [comment, marked as answer]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps

**Reply by inezabonte** [reply]
@leerob I think it's worth noting in the documentation that Notifications are still a feature flag in Safari. Therefore they have to be enabled first for them to work since they are disabled by default.

**Reply by leerob** [reply]
Not if you are using the Web Push API, as far as I could tell from my testing. I did not need to change any flags.

**Reply by inezabonte** [reply]
@leerob You're right... my bad, I last tested this when the Push Api was still a feature flag. All good. Must have missed when it finally became available.
